### PR TITLE
DDF-2327 Add content uri overwrite protection

### PR DESCRIPTION
**Pull request by brendan-hofmann**
#### What does this PR do?
Adds an `AccessPlugin` that prevents resource URIs in the content store from being overwritten.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr @ahoffer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison
#### How should this be tested?
Ingest a product to the content store, such as via the search UI. Verify that attempting to overwrite the resource URI of the generated metacard is rejected.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2327
#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [x] Update / Add Integration Tests

### CODE REVIEW

**Comment by stustison** [comment]
👍

File: catalog/plugin/catalog-plugin-content-uri/src/test/java/org/codice/ddf/catalog/content/plugin/uri/TestContentUriAccessPlugin.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        existingMetacard = mock(Metacard.class);
+        existingMetacards.put(ID_OR_URI, existingMetacard);
+        contentUriAccessPlugin = new ContentUriAccessPlugin();
+    }
+
+    @Test
+    public void bothUrisNull() throws StopProcessingException {
+        when(updateCard.getResourceURI()).thenReturn(null);
+        when(existingMetacard.getResourceURI()).thenReturn(null);
+        contentUriAccessPlugin.processPreUpdate(input, existingMetacards);
```
Comment (peterhuffer) [inline catalog/plugin/catalog-plugin-content-uri/src/test/java/org/codice/ddf/catalog/content/plugin/uri/TestContentUriAccessPlugin.java]: @brendan-hofmann As we discussed - should make an assert here that the input is the same as the output to protect against potential future changes.

**Comment by peterhuffer** [comment]
LGTM after comment is addressed. 👍

**Comment by emanns95** [comment]
Hero passed :+1:

File: catalog/plugin/catalog-plugin-content-uri/pom.xml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>plugin</artifactId>
+        <groupId>ddf.catalog.plugin</groupId>
+        <version>2.10.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>catalog-plugin-content-uri</artifactId>
+    <name>DDF :: Content :: Core :: URI Plugin</name>
```
Comment (pklinef) [inline catalog/plugin/catalog-plugin-content-uri/pom.xml]: I think this should be `DDF :: Catalog :: Plugin :: Content URI`.

**Comment by pklinef** [comment]
👍

**Comment by emanns95** [comment]
Successfully re-heroed :+1:

**Pull request merged**